### PR TITLE
Fix #23 additional.bundles dependencies are not properly resolved

### LINKED ISSUE #23: additional.bundles dependencies are not properly resolved
additional.bundles dependencies are not properly resolved

this issue is initially reported here: https://www.eclipse.org/lists/tycho-user/msg09020.html
besides the xtext-eclipse and xtext-xtend tests builds with tycho 2.4.0-SNAPSHOT
i can also see the problem with
https://github.com/cdietrich/tych24ddBundlesBug

```
[ERROR] Internal error: java.lang.RuntimeException: org.osgi.framework.BundleException: Bundle org.xtext.example.mydsl.ide cannot be resolved:org.xtext.example.mydsl.ide [109]
[ERROR]   Unresolved requirement: Require-Bundle: org.xtext.example.mydsl
[ERROR]     -> Bundle-SymbolicName: org.xtext.example.mydsl; bundle-version="1.0.0.qualifier"; singleton:="true"
[ERROR]        org.xtext.example.mydsl [108]
[ERROR]          Unresolved requirement: Require-Bundle: org.eclipse.xtext.xtext.generator
[ERROR] -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: org.osgi.framework.BundleException: Bundle org.xtext.example.mydsl.ide cannot be resolved:org.xtext.example.mydsl.ide [109]
  Unresolved requirement: Require-Bundle: org.xtext.example.mydsl
    -> Bundle-SymbolicName: org.xtext.example.mydsl; bundle-version="1.0.0.qualifier"; singleton:="true"
       org.xtext.example.mydsl [108]
         Unresolved requirement: Require-Bundle: org.eclipse.xtext.xtext.generator
```

org.eclipse.xtext.xtext.generator is part of xtext.sdk here in the tp https://github.com/cdietrich/tych24ddBundlesBug/blob/7e0dbd51eed5802857b45b4db499041457c4eb5c/org.xtext.example.mydsl.target/org.xtext.example.mydsl.target.target#L18

interestingly i can also find this in the log
```
[INFO] Fetching org.eclipse.xtext.xbase_2.25.0.v20210301-0909.jar from https://ftp.snt.utwente.nl/pub/software/eclipse/releases/2021-03/202103171000/plugins/ (2,46MB)
[INFO] Fetching org.eclipse.xtext.xtext.generator_2.25.0.v20210301-0843.jar from https://ftp.snt.utwente.nl/pub/software/eclipse/releases/2021-03/202103171000/plugins/ (969,17kB)
[INFO] Fetching org.eclipse.equinox.launcher.win32.win32.x86_64_1.2.100.v20210209-1541.jar.pack.gz from https://mirror.dkm.cz/eclipse/releases/2021-03/202103171000/plugins/ (81,75kB)
```

also the problem is on the ide project, whilst die add. bundles are on the mydsl project

```
[INFO] Computing target platform for MavenProject: org.xtext.example.mydsl:org.xtext.example.mydsl:1.0.0-SNAPSHOT @ /Users/cdietrich/eclipse-workspaces/es2103/org.xtext.example.mydsl.parent/org.xtext.example.mydsl/pom.xml
[INFO] Resolving dependencies of MavenProject: org.xtext.example.mydsl:org.xtext.example.mydsl:1.0.0-SNAPSHOT @ /Users/cdietrich/eclipse-workspaces/es2103/org.xtext.example.mydsl.parent/org.xtext.example.mydsl/pom.xml
[INFO] Resolving class path of MavenProject: org.xtext.example.mydsl:org.xtext.example.mydsl:1.0.0-SNAPSHOT @ /Users/cdietrich/eclipse-workspaces/es2103/org.xtext.example.mydsl.parent/org.xtext.example.mydsl/pom.xml
[INFO] Fetching 202103171000&countryCode=de&timeZone=1&format=xml from http://www.eclipse.org/downloads/download.php?format=xml&file=/releases/2021-03/

.... tons of more fetches
[INFO] Computing target platform for MavenProject: org.xtext.example.mydsl:org.xtext.example.mydsl.ide:1.0.0-SNAPSHOT @ /Users/cdietrich/eclipse-workspaces/es2103/org.xtext.example.mydsl.parent/org.xtext.example.mydsl.ide/pom.xml
[INFO] Resolving dependencies of MavenProject: org.xtext.example.mydsl:org.xtext.example.mydsl.ide:1.0.0-SNAPSHOT @ /Users/cdietrich/eclipse-workspaces/es2103/org.xtext.example.mydsl.parent/org.xtext.example.mydsl.ide/pom.xml
[INFO] Resolving class path of MavenProject: org.xtext.example.mydsl:org.xtext.example.mydsl.ide:1.0.0-SNAPSHOT @ /Users/cdietrich/eclipse-workspaces/es2103/org.xtext.example.mydsl.parent/org.xtext.example.mydsl.ide/pom.xml
```

### PR DESCRIPTION
**Pull request by laeubi**
Signed-off-by: Christoph Läubrich <christoph@laeubi-soft.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
do we have Jenkins integration with Github already in place?

File: tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/EquinoxResolver.java [hunk 1]
```diff
@@ -303,13 +303,14 @@ protected void cleanup(ModuleRevision revision) {
                     Collection<String> additionalBundles = buildPropertiesParser.parse(mavenProject.getBasedir())
                             .getAdditionalBundles();
                     if (additionalBundles.size() > 0) {
-                        ArrayList<Object> reqb = new ArrayList<>();
+                        List<String> reqb = new ArrayList<>();
                         String value = mf.getValue(Constants.REQUIRE_BUNDLE);
                         if (value != null) {
                             reqb.add(value);
```
Comment (cdietrich) [inline tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/EquinoxResolver.java:309]: is this one already comma separated?

Comment (laeubi) [reply tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/EquinoxResolver.java:309]: yep

**Comment by laeubi** [comment]
> do we have Jenkins integration with Github already in place?

Yes it just takes some time until the build notice the new PR

**Comment by cdietrich** [comment]
do we have unit or integ tests for such things?

**Comment by laeubi** [comment]
See https://github.com/eclipse/tycho/tree/master/tycho-its for example.

**Pull request merged**